src/highlighter.rs

```diff
@@ -0,0 +1,80 @@
+use {
+    nu_ansi_term::{Color, Style},
+    std::{borrow::Cow, io::Write},
+};
+
+pub static DEFAULT_BUFFER_MATCH_COLOR: Color = Color::Green;
+pub static DEFAULT_BUFFER_NEUTRAL_COLOR: Color = Color::White;
+pub static DEFAULT_BUFFER_NOTMATCH_COLOR: Color = Color::Red;
+
+pub trait Highlighter {
+    fn highlight_external_command<'l>(&self, line: &'l str) -> Cow<'l, str>;
```

Comment:
how about `highlight`? Making it more general so anyone could implement Highlighter and configure it to highlight other things also

Reply:
The initial idea was to have a function for each type of highlighting but we can discuss it.